king_phisher/server/fs_utilities.py

```diff
@@ -107,3 +106,59 @@ def chown(path, user=None, group=constants.AUTOMATIC, recursive=True):
 		iterator = (path,)
 	for path in iterator:
 		os.chown(path, uid, gid)
+
+
+def access(path, mode, user=constants.AUTOMATIC, group=constants.AUTOMATIC):
+	"""
+	This is a high-level wrapper around :py:func:`os.access` to provide
+	additional functionality. Similar to `os.access` this high-level wrapper
+	will test the given path for a variety of access modes. Additionally however,
+	*user* or *group* can be specified to test against a specific of user or group.
+	.. versionadded:: 1.14.0
+	:param str path: The path to test access for.
+	:param mode: The mode to test access for. Set to `os.W_OK` to test for readability,
+		`os.W_OK` for writability and
+		`os.W_OK` to determine if path can be executed.
+	:type mode: :py:class`os.*_OK`
+	:param user: The user to test permissions for. If set to
+		:py:class:`~king_phisher.constants.AUTOMATIC`, the processes current uid
+		will be used.
+	:type user: int, str, ``None``, :py:class:`~king_phisher.constants.AUTOMATIC`
+	:param group: The group to test permissions for. If set to
+		:py:class:`~king_phisher.constants.AUTOMATIC`, the group that *user*
+		belongs too will be used.
+	:type group: int, str, ``None``, :py:class:`~king_phisher.constants.AUTOMATIC`
+	:return: Returns ``True`` only if the user or group has the mode of permission specified else returns ``False``.
+	:rtype: bool
+	"""
+	uid, gid = _resolve_target_ids(user, group)
+	file_info = os.stat(path)
+
+	if file_info.st_uid == uid:
+		# start with no test flags
+		test_mode = 0
+		# then check if R_OK is to be checked for
+		if mode & os.R_OK:
+			# and if so, add the proper flag to be tested
+			test_mode |= stat.S_IRUSR
+		if mode & os.W_OK:
+			test_mode |= stat.S_IWUSR
+		if mode & os.X_OK:
+			test_mode |= stat.S_IXUSR
+		if file_info.st_mode & test_mode:
+			return True
+
+	if isinstance(user, str) and group == constants.AUTOMATIC:
```

Comment:
If `group == constants.AUTOMATIC` and `user` is not a string, but `uid` was resolved, then you should resolve the username for that `uid`. Definitely check if you already have the string to avoid another call to libc if you can, but otherwise you'll have to resolve it yourself.

If `group` is `constants.AUTOMATIC` and `uid` is not an integer, then something horrible happened.

Reply:
Forgot to mention, if necessary resolve the `uid` to a username with [`pylibc.pwgetuid`](https://king-phisher.readthedocs.io/en/latest/king_phisher/server/pylibc.html#king_phisher.server.pylibc.getpwuid).